webpack/assets/javascripts/services/statisticsChartService.js

```diff
@@ -21,6 +22,37 @@ function getChartConfig(details) {
   return config;
 }
 
+function getChartConfigByData(_data, id, type) {
+  const filteredData = _data.filter(d => d[1] !== 0);
+  const data = filteredData.reduce(
+    (curr, next) => {
+      const key = next[0];
+      const names = {
+        ...curr.names,
+        [key]: key.replace(/_/g, ' ')
+      };
+
+      const colors = {
+        ...curr.colors,
+        [key]: next[2]
```

Comment:
whats next[2]? can you add a test so its a bit more understood? overall I'm not sure i understand what this code is doing and why its needed, thanks!